generator/src/Pos/Generator/Block/Mode.hs

```diff
@@ -146,7 +146,7 @@ type BlockGenMode ext m = ReaderT (BlockGenContext ext) m
 type BlockGenRandMode ext g m = RandT g (BlockGenMode ext m)
 
 instance MonadThrow m => MonadThrow (RandT g m) where
-    throwM = lift . throwM
+    throwM = lift . UnsafeExc.throwM
```

Comment:
Wait for it, so you define safe `MonadThrow` instance using lifted unsafe `throwM`. For what reason?

Reply:
I define `MonadThrow.throwM` via `MonadThrow.throwM`. The async exception handling logic is defined separately in `safe-exceptions`.